api/relations/v1/common.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package api.relations.v1;
+
+option go_package = "ciam-rebac/api/relations/v1;v1";
+option java_multiple_files = true;
+option java_package = "api.relations.v1";
+
+message Relationship {
+	ObjectReference resource = 1;
+	string relation = 2;
+	SubjectReference subject = 3;
+}
+
+message SubjectReference {
+	optional string relation = 1;
+	ObjectReference subject = 2;
```

Comment:
Another option here is to separate out a SubjectSet type. In which case a Relationship would look like...

```
message Relationship {
	ObjectReference resource = 1;
	string relation = 2;
	oneof {
		ObjectReference subject = 3;
		SubjectSet subject_set = 4;
}
```

Reply:
Interesting.

Or
```
message Relationship {
	ObjectReference resource = 1;
	string relation = 2;
	oneof {
		SubjectReference subject = 3;
		SubjectSet subject_set = 4;
	}
}
```
where SubjectReference is redefined to be the same as ObjectReference. Duplication, yes, but maybe easier for users to grasp.

Re SubjectSet, how would that look? I suppose it could be identical to what SubjectReference is now.

In any case, it might need to be spiked to see what the resulting code looks like. Not sure how "oneof" is going to translate into a type (but if it's in the spec something sensible obviously happens).